### PR TITLE
Tari Node Controller

### PR DESCRIPTION
**Pull request by StriderDM**
## Description
Implementation of Tari Node Controller.

Currently can be built for Mac & Linux.

Windows support pending, run into issues building GRPC with VS2019.

Screen Shot 2020-10-15 at 4 51 05 PM

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/grpc/base_node_grpc_server.rs [hunk 1]
```diff
@@ -637,6 +646,43 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
...
+        _request: Request<tari_rpc::Empty>,
+    ) -> Result<Response<tari_rpc::SyncInfoResponse>, Status>
+    {
+        debug!(target: LOG_TARGET, "Incoming GRPC request for BN sync data");
+
+        let mut channel = self.state_machine.get_status_info_watch();
+
+        let mut sync_info: Option<BlockSyncInfo> = None;
+
+        if let Some(info) = channel.recv().await {
```
Comment (stringhandler) [inline applications/tari_base_node/src/grpc/base_node_grpc_server.rs:662]: I think this needs to be: 
```suggestion
        if let Some(info) = channel.borrow().await {
```

Comment (StriderDM) [reply applications/tari_base_node/src/grpc/base_node_grpc_server.rs:662]: Says it is unimplemented.

**Comment by delta1** [comment]
Cool! What prerequisites do I need to build/install?

**Comment by StriderDM** [comment]
@delta1 There is a ReadMe.

**Comment by delta1** [comment]
Thanks! sorry I missed that

**Comment by stringhandler** [comment]
I get this error when I try compile with:
```
qmake
make
```
```
In file included from basenode.cpp:1:
In file included from ./basenode.h:6:
./grpcclient.h:4:10: fatal error: 'grpc++/grpc++.h' file not found
#include <grpc++/grpc++.h>
         ^~~~~~~~~~~~~~~~~
```

**Comment by StriderDM** [comment]
@mikethetike Did you build and install grpc from source before you tried to build it?

By defaults:
It should have put the headers into:
```
/usr/local/include 
```
And the libraries into:
```
/usr/local/lib
```

**Comment by stringhandler** [comment]
This should be reopened on https://github.com/tari-project/tari_node_controller

**Pull request closed without merging**